Packet++/header/GvcpLayer.h

```diff
@@ -0,0 +1,772 @@
+	{
+	public:
+		/**
+		 * @brief Construct a new GvcpLayer object
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		GvcpRequestLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		/**
+		 * @brief Construct a new GvcpRequestLayer object
+		 * @param[in] data A pointer to the data including the header and the payload
+		 * @param[in] dataSize The size of the data in bytes
+		 */
+		GvcpRequestLayer(const uint8_t* data, size_t dataSize);
+
+		/**
+		 * @brief Construct a new GvcpRequestLayer object
+		 * @param[in] command The command
+		 * @param[in] payloadData A pointer to the payload data, optional
+		 * @param[in] payloadDataSize The size of the payload data in bytes, optional
+		 * @param[in] flag The flag, optional
+		 * @param[in] requestId The request ID, it should be always larger than 1, optional
+		 * @note all the parameters will be converted to the network byte order
+		 */
+		explicit GvcpRequestLayer(GvcpCommand command, const uint8_t* payloadData = nullptr,
+		                          uint16_t payloadDataSize = 0, GvcpFlag flag = 0, uint16_t requestId = 1);
+
+		/**
+		 * @brief Get the header object
+		 * @return GvcpRequestHeader* A pointer to the header object
+		 */
+		GvcpRequestHeader* getGvcpHeader() const
+		{
+			return reinterpret_cast<GvcpRequestHeader*>(m_Data);  // the header is at the beginning of the data
+		}
+
+		/**
+		 * @brief Get the flag from the header
+		 */
+		GvcpFlag getFlag() const
+		{
+			return getGvcpHeader()->flag;
+		}
+
+		/**
+		 * @brief Get the data size from the header
+		 */
+		uint16_t getDataSize() const
+		{
+			return netToHost16(getGvcpHeader()->dataSize);
+		}
+
+		/**
+		 * @brief Get the request ID from the header
+		 */
+		uint16_t getRequestId() const
+		{
+			return netToHost16(getGvcpHeader()->requestId);
+		}
+
+		/**
+		 * @brief Get the command from the header
+		 */
+		GvcpCommand getCommand() const
+		{
+			return static_cast<GvcpCommand>(netToHost16(getGvcpHeader()->command));
+		}
+
+		/**
+		 * @brief Verify the magic number in the header
+		 * @return true The magic number is valid
+		 */
+		bool verifyMagicNumber() const
+		{
+			return getGvcpHeader()->magicNumber == internal::kGvcpMagicNumber;
+		}
+
+		/**
+		 * @brief Check if the acknowledge is required from the header
+		 * @return true The acknowledge is required
+		 */
+		bool hasAcknowledgeFlag() const
+		{
+			constexpr GvcpFlag kAcknowledgeFlag = 0b0000001;
+			return (getGvcpHeader()->flag & kAcknowledgeFlag) == kAcknowledgeFlag;
+		}
+
+		// implement Layer's abstract methods
+		std::string toString() const override;
+
+		// implement Layer's abstract methods
+		size_t getHeaderLen() const override
+		{
+			return sizeof(GvcpRequestHeader);
+		}
```

Comment:
The header length should include the entire layer length - in this case since GVCP layer is always last, it should return `m_DataLen` for all type of GVCP messages

Reply:
I don't get it. The header length should not cover the payload. GVCP layer includes GVCP header and GVCP payload. In this case, why is `getHeaderLen()`  not the GVCP header length?